allennlp/common/file_utils.py

```diff
@@ -243,6 +242,46 @@ def _find_latest_cached(url: str, cache_dir: str) -> Optional[str]:
     return None
 
 
+class CacheFile:
+    """
+    This is a context manager that makes robust caching easier.
+
+    On `__enter__`, an IO handle to a temporarily file is returned, which can
+    be treated as if it's the actual cache file.
+
+    On `__exit__`, the temporarily file is renamed to the cache file. If anything
+    goes wrong while writing to the temporary file, it will be removed.
+    """
+
+    def __init__(self, cache_filename: Union[Path, str], mode="w+b") -> None:
```

Comment:
The default file mode should probably be whatever the default is in `open()`, which is `t`, not `b`. Just to minimize surprise.

Reply:
`w+b` is the default for `NamedTemporaryFile` though. I'm fine with changing it, but if we're consistent with one we're inconsistent with the other

Reply:
That seems like an oversight in Python then. But yeah, I agree, let's be consistent with `NamedTemporaryFile` then.